src/createAnimatedComponent/JSPropUpdater.ts

```diff
@@ -4,14 +4,20 @@ import {
   NativeModules,
   findNodeHandle,
 } from 'react-native';
-import { nativeShouldBeMock } from '../reanimated2/PlatformChecker';
+import { isWeb, nativeShouldBeMock } from '../reanimated2/PlatformChecker';
 import type { StyleProps } from '../reanimated2';
 
 interface ListenerData {
   viewTag: number;
   props: StyleProps;
 }
 
+interface WebAnimatedComponent extends React.Component<unknown, unknown> {
```

Comment:
I'm also not sure about that, but I used it because it is already done in this file (e..g [here](https://github.com/software-mansion/react-native-reanimated/blob/7ce3e3e1b1ba7233602f6e740c116cf257a7b9b4/src/createAnimatedComponent/JSPropUpdater.ts#L43))

Reply:
Okay, we'll need to take a look on types but in a separate PR, thanks for pointing this out!

Reply:
Yep, but you're introducing a new interface `WebAnimatedComponent` in this PR.

Reply:
Oh, now I see, you're absolutely right! Done in [58b23fa](https://github.com/software-mansion/react-native-reanimated/pull/5300/commits/58b23faf61c17186ba28c547117fd12969e8f3da).

Reply:
Okay, I think [b5903b7](https://github.com/software-mansion/react-native-reanimated/pull/5300/commits/b5903b7a60d78f15cb76d58223c374151f792cdb) makes it look much better

Reply:
I think it looks good for now (I only took a glance)